packages/rulesets/src/spectral/az-common.ts

```diff
@@ -307,6 +312,99 @@ const ruleset: any = {
         function: xmsPathsMustOverloadPaths,
       },
     },
+    XmsExamplesRequired: {
+      description: 'Verifies whether `x-ms-examples` are provided for each operation or not.',
+      message: 'Please provide x-ms-examples describing minimum/maximum property set for response/request payloads for operations.',
+      severity: "warn",
+      resolved: false,
+      formats: [oas2],
+      given: ["$.paths.*[get,put,post,patch,delete,options,head]"],
+      then: {
+        function: xmsExamplesRequired
+      }
+    },
+    XmsClientNameParameter: {
+      description:
+          'The `x-ms-client-name` extension is used to change the name of a parameter or property in the generated code. ' +
+          'By using the `x-ms-client-name` extension, a name can be defined for use specifically in code generation, separately from the name on the wire. ' +
+          'It can be used for query parameters and header parameters, as well as properties of schemas. This name is case sensitive.',
+      message:
+          'Value of `x-ms-client-name` cannot be the same as Property/Model.',
+      severity: "warn",
+      resolved: false,
+      formats: [oas2],
+      given: ["$.paths.*[get,put,post,patch,delete,options,head].parameters[?(@.name && @['x-ms-client-name'])]","$.paths.*.parameters[?(@.name && @['x-ms-client-name'])]","$.parameters.[?(@.name && @['x-ms-client-name'])]"],
+      then: {
+        function: xmsClientNameParameter
+      }
+    },
+    XmsClientNameProperty: {
+      description:
+          'The `x-ms-client-name` extension is used to change the name of a parameter or property in the generated code.' +
+          'By using the `x-ms-client-name` extension, a name can be defined for use specifically in code generation, separately from the name on the wire.' +
+          'It can be used for query parameters and header parameters, as well as properties of schemas. This name is case sensitive.',
+      message:
+          'Value of `x-ms-client-name` cannot be the same as Property/Model.',
+      severity: "warn",
+      resolved: false,
+      formats: [oas2],
+      given: ["$.definitions[*].properties.*['x-ms-client-name']"],
+      then: {
+        function: xmsClientNameProperty
+      }
+    },
+    ListInOperationName: {
+      description: 'Verifies whether value for `operationId` is named as per ARM guidelines when response contains array of items.',
+      message: 'Since operation response has model definition, it should be of the form "_list".',
+      severity: "warn",
+      resolved: false,
+      formats: [oas2],
+      given: ["$.paths[*].get['x-ms-pageable']^.operationId"],
```

Comment:
seems in the original rule, the operation whose response is array as well is also selected